Core/Tornado/Server/TornadoService.py

```diff
@@ -0,0 +1,526 @@
+    """
+    self.debug = debug
+    self.authorized = False
+    self.method = None
+    self.requestStartTime = time.time()
+    self.credDict = None
+    self.authorized = False
+    self.method = None
+
+    # On internet you can find "HTTP Error Code" or "HTTP Status Code" for that.
+    # In fact code>=400 is an error (like "404 Not Found"), code<400 is a status (like "200 OK")
+    self._httpError = HTTPErrorCodes.HTTP_OK
+    if not self.__FLAG_INIT_DONE:
+      init = self.__initializeService(self.srv_getURL(), self.request.full_url(), debug)
+      if not init['OK']:
+        self._httpError = HTTPErrorCodes.HTTP_INTERNAL_SERVER_ERROR
+        gLogger.error("Error during initalization on %s" % self.request.full_url())
+        gLogger.debug(init)
+        return False
+
+    self._stats['requests'] += 1
+    self._monitor.setComponentExtraParam('queries', self._stats['requests'])
+    self._monitor.addMark("Queries")
+    return True
+
+  def prepare(self):
+    """
+      prepare the request, it read certificates and check authorizations.
+    """
+    self.method = self.get_argument("method")
+    self.rawContent = self.get_argument('rawContent', default=False)
+    self.log.notice("Incoming request on /%s: %s" % (self._serviceName, self.method))
+
+    # Init of service must be checked here, because if it have crashed we are
+    # not able to end request at initialization (can't write on client)
+    if not self.__FLAG_INIT_DONE:
+      error = encode("Service can't be initialized ! Check logs on the server for more informations.")
+      self.__write_return(error)
+      self.finish()
+
+    try:
+      self.credDict = self._gatherPeerCredentials()
+    except Exception:  # pylint: disable=broad-except
+      # If an error occur when reading certificates we close connection
+      # It can be strange but the RFC, for HTTP, say's that when error happend
+      # before authenfication we return 401 UNAUTHORIZED instead of 403 FORBIDDEN
+      self.reportUnauthorizedAccess(HTTPErrorCodes.HTTP_UNAUTHORIZED)
+
+    try:
+      hardcodedAuth = getattr(self, 'auth_' + self.method)
+    except AttributeError:
+      hardcodedAuth = None
+
+    self.authorized = self._authManager.authQuery(self.method, self.credDict, hardcodedAuth)
+    if not self.authorized:
+      self.reportUnauthorizedAccess()
+
+  @gen.coroutine
+  def post(self):  # pylint: disable=arguments-differ
+    """
+    HTTP POST, used for RPC
+      Call the remote method, client may send his method via "method" argument
+      and list of arguments in JSON in "args" argument
+    """
+
+    # Execute the method
+    # First argument is "None", it's because we let Tornado manage the executor
+    retVal = yield IOLoop.current().run_in_executor(None, self.__executeMethod)
+
+    # Tornado recommend to write in main thread
+    # TODO CHRIS READ THAT https://www.tornadoweb.org/en/branch5.1/web.html#thread-safety-notes
```

Comment:
Reminder to decide if you want to address this before or after merging (same applies to the other `TODO CHRIS` comments)

Reply:
right ! :-)